[ProfilePage] Adds user profile page route

GustavJaner opened the pull request:
# Description
- Adds user profile page route
- Passes id of the user with route props: path=/profile/:id

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+  };
+
+  async componentDidMount(props) {
+    const path = "/videos/" + this.props.id;
```
You can use string interpolation and destructuring to make this prettier:
```
const { id } = this.props;
const data = await API.get('videocloud', `/videos/${id}`)';
```

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      this.setState({ video: data });
+      this.s3loader();
+
+      console.log("state.video: ", this.state.video);
```
I think you left this here by accident.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+  }
+
+  async s3loader(props) {
+    // Storage.configure({ level: 'private' });
```
Delete this commented line.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      const url = await Storage.get(this.state.video.location, { level: 'private' })
+      this.setState({ S3url: url });
+
+      console.log("state.S3url: ", this.state.S3url);
```
Delete this debugging line.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+
+      console.log("state.S3url: ", this.state.S3url);
+    } catch (error) {
+      console.log(error);
```
You can print the error, but use `console.error(...)`

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+    }
+  }
+
+  loader(props) {
```
`props` is not used.

edgarlpassos commented on src/containers/Home.jsx:
```diff
@@ -1,11 +1,15 @@
       <Container className="home-container" fluid>
         <h1>Home Page</h1>
+        <hr className="mt-1 mb-5" />
+        <VideoFeed numVideos="3" owner="36" />
```
Why are these values hardcoded?

edgarlpassos commented on src/containers/Profile.jsx:
```diff
@@ -0,0 +1,29 @@
+      <Container className="Profile-container" fluid>
+        <h1>Profile Page { this.state.user }</h1>
+        <hr className="mt-1 mb-5" />
+        <VideoFeed numVideos="3" owner={ this.state.user } />
```
Same thing about the hardcoded values.

edgarlpassos reviewed (changes requested):
Don't forget to squash the commits in the PR.

GustavJaner committed:
[ProfilePage] Adds user profile page route
- Passes id of the user with route prop: path=/profile/:id
- Passes number of videos to render with the optional route prop: path=/profile/:id/:numVids?
